aspnet/mvc/models/formatting.rst

```diff
@@ -1,8 +1,223 @@
+
+To return data using a specific result type, assuming you're inheriting from the base ``Controller`` class, it's recommended that you use the built-in helper method ``Json`` to return JSON and ``Content`` plain text. These methods check to see if the data being formatted implements ``IDisposable`` and register the object for disposal if required. Your action method should return either the specific result type (for instance, ``JsonResult``) or ``IActionResult``.
+
+Returning JSON-formatted data:
+
+.. literalinclude:: formatting/sample/src/ResponseFormattingSample/Controllers/Api/AuthorsController.cs
+  :language: c#
+  :lines: 21-26
+  :emphasize-lines: 3,5
+  :dedent: 8
+
+Sample response from this action:
+
+.. image:: formatting/_static/json-response.png
+
+Note that the content type of the response is ``application/json``, shown both in the list of network requests and in the Response Headers section. Also note the list of options presented by the browser (in this case, Microsoft Edge) in the Accept header, in the Request Headers section. The current technique is ignoring this header; obeying it is discussed below.
+
+To return plain text formatted data, use ``ContentResult`` and the ``Content`` helper:
+
+.. literalinclude:: formatting/sample/src/ResponseFormattingSample/Controllers/Api/AuthorsController.cs
+  :language: c#
+  :lines: 47-52
+  :emphasize-lines: 3,5
+  :dedent: 8
+
+A response from this action:
+
+.. image:: formatting/_static/text-response.png
+
+Note in this case the ``Content-Type`` returned is ``text/plain``. You can also achieve this same behavior using just a string response type:
+
+.. literalinclude:: formatting/sample/src/ResponseFormattingSample/Controllers/Api/AuthorsController.cs
+  :language: c#
+  :lines: 54-59
+  :emphasize-lines: 3,5
+  :dedent: 8
+
+.. tip:: For non-trivial actions with multiple return types or options (for example, different HTTP status codes based on the result of operations performed), prefer ``IActionResult`` as the return type.
+
+Content Negotiation
+-------------------
+
+Content negotiation (*conneg* for short) occurs when the client specifies an `Accept header <https://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html>`_. The default format used by ASP.NET Core MVC is JSON. Negotiation occurs automatically for ``IActionResult`` return types, including results returned using the helper methods ``Ok``, ``BadRequest``, ``Created``, etc. You can also return a model type (a class you've defined as your data transfer type) and the framework will automatically wrap it in an ``ObjectResult`` for you.
```

Comment:
All status code specific results are based on ObjectResult. We shouldn't associate IActionResult with content negotiation, as its not the case.